citizen.ts

```diff
@@ -344,3 +391,125 @@ export function scheduleRendezvous(
 
   return `${targetContractSource}\n\n${context}\n\n${tests}`;
 }
+
+/**
+ * Utility function that restores the test wallets' initial sBTC balances in
+ * the re-initialized first-class citizenship simnet.
+ *
+ * @param simnet The simnet instance.
+ * @param sbtcBalancesMap A map containing the test wallets' balances to be
+ * restored.
+ */
+const restoreSbtcBalances = (
+  simnet: Simnet,
+  sbtcBalancesMap: Map<string, number>
+) => {
+  // To deposit sBTC, one needs a txId and a sweep txId. A deposit transaction
+  // must have a unique txId and sweep txId. A set will be used to track the
+  // already used ids. This set will be used in `getUniqueRandomHex`.
+  const usedTxIds: Set<string> = new Set();
+
+  // For each address present in the balances map, restore the balance.
+  Array.from(sbtcBalancesMap.entries())
+    // Re-assure the map does not contain nil balances.
+    .filter(([_, balance]) => balance !== 0)
+    .forEach(([address, balance]) => {
+      const txId = getUniqueRandomHex(usedTxIds);
+      const sweepTxId = getUniqueRandomHex(usedTxIds);
+      mintSbtc(simnet, balance, address, txId, sweepTxId);
+    });
+};
+
+/**
+ * Utility function to deposit an amount of sBTC to a Stacks address.
+ *
+ * @param simnet The simnet instance.
+ * @param amountSats The amount to mint in sats.
+ * @param recipient The Stacks address to mint sBTC to.
+ * @param txId A unique hex to use for the deposit.
+ * @param sweepTxId A unique hex to use for the deposit.
+ */
+const mintSbtc = (
+  simnet: Simnet,
+  amountSats: number,
+  recipient: string,
+  txId: string,
+  sweepTxId: string
+) => {
+  const blockHeight = simnet.blockHeight;
+
+  // Calling `get-burn-header` only works for past block heights. We mine one
+  // empty Bitcoin block along with a Stacks one to make sure this happens.
+  simnet.mineEmptyBurnBlocks(1);
```

Comment:
`mintSBtc()` is called for every account that needs to be founded, so if there are 10 accounts you'll mine 10 burn blocks, if there are 2 accounts you'll mine 2 burn blocks.
That might be weird.
I think you could simple pass `Cl.uint(simnet.burnBlockHeight - 1)`